tokio-util/Cargo.toml

```
@@ -21,15 +21,16 @@ categories = ["asynchronous"]
 default = []
 
 # Shorthand for enabling everything
-full = ["codec", "compat", "io-util", "time", "net", "rt"]
+full = ["codec", "compat", "io-util", "time", "net", "rt", "join-map"]
 
 net = ["tokio/net"]
 compat = ["futures-io",]
 codec = []
 time = ["tokio/time","slab"]
 io = []
 io-util = ["io", "tokio/rt", "tokio/io-util"]
-rt = ["tokio/rt", "tokio/sync", "futures-util", "hashbrown"]
+rt = ["tokio/rt", "tokio/sync", "futures-util"]
+join-map = ["tokio/rt", "hashbrown"]
```

I think we should just have `join-map` also enable the `rt` feature. After all, the join map does require a runtime.
```suggestion
rt = ["tokio/rt", "tokio/sync", "futures-util"]
join-map = ["rt", "hashbrown"]
```
This way, we can simplify the `task` module to only exist when `rt` is enabled instead of the `rt or join-map` requirement it has today.